### PR TITLE
Pants: add `python_distribution` BUILD metadata for st2* components

### PR DESCRIPTION
**Pull request by cognifloyd**
### Background

This is another part of introducing [pants](https://www.pantsbuild.org/docs), as discussed in various TSC meetings.

Related PRs can be found in:
- [pantsbuild](https://github.com/StackStorm/st2/milestone/47) milestone (which includes PRs since #5713)
- https://github.com/StackStorm/st2/labels/pantsbuild label (which also includes preparatory PRs that came before adding pantsbuild)

### Overview of this PR

This PR adds the `st2_component_python_distribution()` macro to the BUILD metadata for all of our components (`st2common`, `st2client`, `st2auth`, etc).

Thanks to the macro, we only have to define 3 (or 4) parameters for each runner: `component_name`, `scripts`, and `entry_points`. I copied the scripts and entry_points from the `setup.py` file, which will become obsolete once pants can generate the setup.py file for us.

Under `entry_points`, I have "tagged" the `st2auth.sso.backends`, the `st2common.metrics.driver` and the `st2common.rbac.backend` categories as `stevedore_namespace`s. Once we switch to 2.16 and enable the `pants.backend.experimental.python.framework.stevedore` plugin, the `stevedore` plugin will, for tests, use this "tag" to look up which wheels implement a plugin in that stevedore namespace.

To facilitate comparison, I will add a comment to each of the files that shows the current `setup.py` entry_points and scripts.

### Next steps

Follow-up PRs will adjust the dependencies on the `python_distribution`s to ensure all files get included.

### CODE REVIEW

File: st2actions/BUILD [hunk 1]
```diff
@@ -0,0 +1,10 @@
+st2_component_python_distribution(
+    component_name="st2actions",
+    scripts=[
+        "bin/st2actionrunner",
+        "bin/st2notifier",
+        "bin/st2workflowengine",
+        "bin/st2scheduler",
+        "bin/runners.sh:shell",  # used by service files installed by st2-packaging
+    ],
```
Comment (cognifloyd) [inline st2actions/BUILD:9]: https://github.com/StackStorm/st2/blob/f1159344e0ca1630de13e88042589c5e472de2a0/st2actions/setup.py#L49-L54

Plus I noticed that st2-packaging needs runners.sh, so I included that as well.

File: st2api/BUILD [hunk 2]
```diff
@@ -0,0 +1,4 @@
+st2_component_python_distribution(
+    component_name="st2api",
+    scripts=["bin/st2api"],
```
Comment (cognifloyd) [inline st2api/BUILD:3]: https://github.com/StackStorm/st2/blob/f1159344e0ca1630de13e88042589c5e472de2a0/st2api/setup.py#L48

File: st2auth/BUILD [hunk 3]
```diff
@@ -0,0 +1,9 @@
+st2_component_python_distribution(
+    component_name="st2auth",
+    scripts=["bin/st2auth"],
+    entry_points={
+        stevedore_namespace("st2auth.sso.backends"): {
+            "noop": "st2auth.sso.noop:NoOpSingleSignOnBackend",
+        },
+    },
```
Comment (cognifloyd) [inline st2auth/BUILD:8]: https://github.com/StackStorm/st2/blob/f1159344e0ca1630de13e88042589c5e472de2a0/st2auth/setup.py#L49-L52

File: st2client/BUILD [hunk 4]
```diff
@@ -0,0 +1,8 @@
+st2_component_python_distribution(
+    component_name="st2client",
+    entry_points={
+        "console_scripts": {
+            "st2": "st2client.shell:main",
+        },
+    },
```
Comment (cognifloyd) [inline st2client/BUILD:7]: https://github.com/StackStorm/st2/blob/f1159344e0ca1630de13e88042589c5e472de2a0/st2client/setup.py#L69

File: st2common/BUILD [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    entry_points={
+        stevedore_namespace("st2common.metrics.driver"): {
+            "statsd": "st2common.metrics.drivers.statsd_driver:StatsdDriver",
+            "noop": "st2common.metrics.drivers.noop_driver:NoopDriver",
+            "echo": "st2common.metrics.drivers.echo_driver:EchoDriver",
+        },
+        stevedore_namespace("st2common.rbac.backend"): {
+            "noop": "st2common.rbac.backends.noop:NoOpRBACBackend",
+        },
+    },
```
Comment (cognifloyd) [inline st2common/BUILD:38]: https://github.com/StackStorm/st2/blob/f1159344e0ca1630de13e88042589c5e472de2a0/st2common/setup.py#L50-L82

A few of the scripts were missing in setup.py, so I included them here. Also, a couple of them were missing the execute bit, so I added that as well.,

File: st2reactor/BUILD [hunk 6]
```diff
@@ -0,0 +1,11 @@
+st2_component_python_distribution(
+    component_name="st2reactor",
+    scripts=[
+        "bin/st2-rule-tester",
+        "bin/st2-trigger-refire",
+        "bin/st2rulesengine",
+        "bin/st2sensorcontainer",
+        "bin/st2garbagecollector",
+        "bin/st2timersengine",
+    ],
```
Comment (cognifloyd) [inline st2reactor/BUILD:10]: https://github.com/StackStorm/st2/blob/f1159344e0ca1630de13e88042589c5e472de2a0/st2reactor/setup.py#L49-L56

File: st2stream/BUILD [hunk 7]
```diff
@@ -0,0 +1,4 @@
+st2_component_python_distribution(
+    component_name="st2stream",
+    scripts=["bin/st2stream"],
```
Comment (cognifloyd) [inline st2stream/BUILD:3]: https://github.com/StackStorm/st2/blob/f1159344e0ca1630de13e88042589c5e472de2a0/st2stream/setup.py#L48

**Comment by cognifloyd** [comment]
#5906 should be merged first

**Review by amanda11 — COMMENTED** [review]
Why are st2common/bin/st2-purge-task-executions and st2common/bin/st2-purge-workflows now empty files?
Is that github just showing it odd in PR, and all you've done is change the permissions on the file?

**Comment by cognifloyd** [comment]
Yeah. That's just GitHub being weird as I only changed the execute bit, not the file contents. If you do a "view file" on those, you'll see the file contents.

**Review by amanda11 — APPROVED** [review]

**Pull request merged**